pandas/tests/indexes/test_multi.py

```diff
@@ -154,7 +154,7 @@ def assert_matching(actual, expected):
             # as much as possible
             self.assertEqual(len(actual), len(expected))
             for act, exp in zip(actual, expected):
-                act = np.asarray(act)
+                act = np.asarray(act, dtype=np.object_)
```

Comment:
Just pushed up commit https://github.com/pydata/pandas/pull/14236/commits/e0c8897a33dc480f64cc221f71dd095aa97e0fd8 that undoes this change and fails with the error:
```
E       AssertionError: numpy array are different
E       
E       Attribute "dtype" are different
E       [left]:  int8
E       [right]: object
```